app/assets/v2/js/pages/new_bounty.js

```diff
@@ -197,139 +125,119 @@ $(document).ready(function(){
-                }
-            } else {
-                var url = etherscan_tx_url(result);
-                var msg = "We've just submited the <a href="+url+" target=new>first transaction to the blockchain</a>.  Hang tight for a few seconds (can sometimes take up to a minute depending upon gas settings & network state) while it confirms.";
-                _alert({ title: "Transaction #1 Submitted", message: msg  }, 'info');
-                next();
+                _alert({ message: "There was an error.  Please try again or contact support." });
+                $('#submitBounty').removeAttr('disabled');
+                return;
             }
-        };
-        var sendERC20Approve = function(){
-            token_contract.approve.estimateGas(bounty_address()
-                ,amount, 
-                function(errors,result){
-                    mixpanel.track("New Bounty Error", {step: 'erc20', error: errors});
-                    var gas = Math.round(erc20_approve_gas * gasMultiplier);
-                    var gasLimit = Math.round(gas * gasLimitMultiplier);
-                    token_contract.approve.sendTransaction(bounty_address()
-                        ,amount, 
-                        {from :account, 
-                            gas:web3.toHex(gas), 
-                            gasLimit: web3.toHex(gasLimit), 
-                            gasPrice:web3.toHex($("#gasPrice").val() * 10**9), 
-                        },
-                        erc20_approve_callback);
-                });
-        };
-        // actually send the transactions to web3
-        setTimeout(function(){
-            bounty.bountydetails.call(issueURL, function(error, result){
-                if(error){
-                    console.error(error);
-                    mixpanel.track("New Bounty Error", {step: 'details', error: error});
-                    _alert({ message: "There was an error.  Please try again or contact support." });
-                    unloading_button($('#submitBounty'));
-                    return;
-                }
-                var isOpenAlready = result[4];
-                if(isOpenAlready){
-                    _alert("There is already an open funding on this issue.  Please try again with another issue.");
-                    unloading_button($('#submitBounty'));
-                    return;
-                }
-                if(!isETH){
-                    sendERC20Approve();
-                } else {
-                    sendPostBounty();
-                }
-            });
-        },100);
-        e.preventDefault();
-    });
+            localStorage['dataHash'] = result;  // cache data hash to find bountyId later
+            // bounty is a web3.js eth.contract address
+            // The Ethereum network requires using ether to do stuff on it
+            // issueAndActivateBounty is a method definied in the StandardBounties solidity contract.
+            var bountyIndex = bounty.issueAndActivateBounty(
```

Comment:
i think that the issueAndActivateBounty should be wrapped in an `.estimateGas()` method... so that way we know what gas / gaslimit to pass into metamask... no? 

estimategas returns `errors, gas`, where gas will be an integer we can use to do this.

Reply:
this is actually really important for making new issue submissions as cheap as possible gas-wise.